src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/data/Card.java

```diff
@@ -0,0 +1,49 @@
+package com.google.step.YOUR_PROJECT_NAME_HERE.data;
+
+public final class Card {
+  private String title;
+  private String code;
+  private String link;
+  private String description;
+
+  public Card() {}
+
+  public Card(String title, String code, String link, String description) {
+    this.title = title;
+    this.code = code;
+    this.link = link;
+    this.description = description;
+  }
+
+  public void setTitle(String title) {
+    this.title = title;
+  }
+
+  public String getTitle() {
+    return this.title;
+  }
+
+  public void setCode(String code) {
+    this.code = code;
+  }
+
+  public String getCode() {
+    return this.code;
+  }
+
+  public void setLink(String link) {
+    this.link = link;
+  }
+
+  public String getLink() {
+    return this.link;
+  }
+
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  public String getDescription() {
+    return this.description;
```

Comment:
If you can return the values rather than set them in the helper functions themselves, we can actually probably avoid having to use these helper functions.

Reply:
Because also, I don't think getDescription/getLink/getTitle are used anywhere else in your code

Reply:
need to use getCode() to store the question_id, answer_id before actually being able to get the code body

Reply:
Can you pass in the question_id, answer_id as arguments for your function that retrieves the code body rather than storing them in the code field of the card via getCode()?